Knossos.NET/Classes/FsoBuild.cs

```diff
@@ -1056,6 +1016,64 @@ private int DetermineScore(string modpath)
                         break;
                 }
             }
+            return score;
+        }
+    }
+
+    public class FsoFile
+    {
+        public string filename;
+        public FsoExecType type;
+        public FsoExecArch arch;
+        public FsoExecEnvironment env;
+        public bool isValid = false;
+        public int score { get; internal set; } = 0;
+
+        public FsoFile(string filename, string modpath, FsoExecType type, FsoExecArch arch, FsoExecEnvironment env)
+        {
+            this.filename = filename;
+            this.type = type;
+            this.arch = arch;
+            this.env = env;
+            this.score = DetermineScore(modpath);
+            if(score > 0)
+                isValid = true;
+        }
+
+        /// <summary>
+        /// Determine FSO File score based on OS and CPU Arch
+        /// </summary>
+        /// <param name="modpath"></param>
+        /// <returns>int score from 0 to 100</returns>
+        private int DetermineScore(string modpath)
+        {
+            int score = 0;
+            string filePath = string.Empty;
+            if (modpath != string.Empty)
+                filePath = Path.Combine(modpath, filename);
+            /* First the cases that are an instant 0 */
+            if (arch == FsoExecArch.other || env == FsoExecEnvironment.Unknown || type == FsoExecType.Unknown)
+            {
+                if (modpath != string.Empty)
+                    Log.Add(Log.LogSeverity.Warning, "FsoFile.DetermineScore", "File: " + filePath + " has an unknown cpu arch, build or enviroment type in json.");
+                return 0;
+            }
+            // get exe name here for exists check
+            if (modpath != string.Empty && !File.Exists(filePath))
+            {
+                if (modpath != string.Empty)
+                    Log.Add(Log.LogSeverity.Warning, "FsoFile.DetermineScore", "File: " + filePath + " does not exist!");
+                return 0;
+            }
+
+            if (env == FsoExecEnvironment.Windows && !KnUtils.IsWindows || env == FsoExecEnvironment.Linux && !KnUtils.IsLinux || env == FsoExecEnvironment.MacOSX && !KnUtils.IsMacOS)
```

Comment:
I would think that you would need to have parentheses in this conditional for it to work.

For example: 
```if ((env == FsoExecEnvironment.Windows && !KnUtils.IsWindows) || (env == FsoExecEnvironment.Linux && !KnUtils.IsLinux) || (env == FsoExecEnvironment.MacOSX && !KnUtils.IsMacOS))```


Is that just a difference between C# and C++?

Reply:
And to be fair VS dosent say they are unnecessary so...